hera_mc/cm_sysdef.py

```diff
@@ -6,123 +6,229 @@
-    print("\t\tFor this part type:  'parts.py conn -p {}'".format(part.hpn))
-    print("\t\tThen re-run hookup with a csv list of parts.")
-
-
-def get_port_pols_to_do(part, port_query):
+import six
+from argparse import Namespace
+
+port_def = {}
+port_def['parts_hera'] = {
+    'station': {'up': [[None]], 'down': [['ground']], 'position': 0},
+    'antenna': {'up': [['ground']], 'down': [['focus']], 'position': 1},
+    'feed': {'up': [['input']], 'down': [['terminals']], 'position': 2},
+    'front-end': {'up': [['input']], 'down': [['e'], ['n']], 'position': 3},
+    'cable-rfof': {'up': [['ea'], ['na']], 'down': [['eb'], ['nb']], 'position': 4},
+    'post-amp': {'up': [['ea'], ['na']], 'down': [['eb'], ['nb']], 'position': 5},
+    'snap': {'up': [['e2', 'e6', 'e10'], ['n0', 'n4', 'n8']], 'down': [['rack']], 'position': 6},
+    'node': {'up': [['loc0', 'loc1', 'loc2', 'loc3']], 'down': [[None]], 'position': 7}
+}
+port_def['parts_paper'] = {
+    'station': {'up': [[None]], 'down': [['ground']], 'position': 0},
+    'antenna': {'up': [['ground']], 'down': [['focus']], 'position': 1},
+    'feed': {'up': [['input']], 'down': [['terminals']], 'position': 2},
+    'front-end': {'up': [['input']], 'down': [['e'], ['n']], 'position': 3},
+    'cable-feed75': {'up': [['ea'], ['na']], 'down': [['eb'], ['nb']], 'position': 4},
+    'cable-post-amp(in)': {'up': [['a']], 'down': [['b']], 'position': 5},
+    'post-amp': {'up': [['ea'], ['na']], 'down': [['eb'], ['nb']], 'position': 6},
+    'cable-post-amp(out)': {'up': [['a']], 'down': [['b']], 'position': 7},
+    'cable-receiverator': {'up': [['a']], 'down': [['b']], 'position': 8},
+    'cable-container': {'up': [['a']], 'down': [['b']], 'position': 9},
+    'f-engine': {'up': [['input']], 'down': [[None]], 'position': 10}
+}
+port_def['parts_rfi'] = {
+    'station': {'up': [[None]], 'down': [['ground']], 'position': 0},
+    'antenna': {'up': [['ground']], 'down': [['focus']], 'position': 1},
+    'feed': {'up': [['input']], 'down': [['terminals']], 'position': 2},
+    'temp-cable': {'up': [['ea'], ['na']], 'down': [['eb'], ['nb']], 'position': 3},
+    'snap': {'up': [['e2', 'e6', 'e10'], ['n0', 'n4', 'n8']], 'down': [['rack']], 'position': 4},
+    'node': {'up': [['loc0', 'loc1', 'loc2', 'loc3']], 'down': [[None]], 'position': 5}
+}
+port_def['parts_test'] = {
+    'vapor': {'up': [[None]], 'down': [[None]], 'position': 0}
+}
+pind = {}
+this_hookup_type = None
+
+
+def sys_init(husys, v0):
+    y = {}
+    for x in husys:
+        y[x] = v0
+    return y
+
+
+checking_order = ['parts_hera', 'parts_rfi', 'parts_paper', 'parts_test']
+
+# Initialize the dictionaries
+corr_index = sys_init(checking_order, None)
+all_pols = sys_init(checking_order, [])
+redirect_part_types = sys_init(checking_order, [])
+single_pol_labeled_parts = sys_init(checking_order, [])
+
+# Redefine dictionary as needed
+corr_index['parts_hera'] = 6
+corr_index['parts_paper'] = 10
+corr_index['parts_rfi'] = 4
```

Comment:
what do these lines do? I'm unclear about why they're needed right after you did an initialization and on what the hardcoded integers mean.

Reply:
cm_sysdef contains all of the "hard-coded" aspects of a system (and only in cm_sysdef!). This is needed to tell which part position is the correlator so cm_corrinfo knows where to look.

Reply:
ah, ok. Maybe add a more descriptive comment?